board/common/rootfs/usr/libexec/infix/init.d/05-product

```diff
@@ -1,22 +1,44 @@
 #!/bin/sh
-# Find and install any product specific files in /etc before bootstrap
+# Find, install, and run product specific files and script in /etc
+# before resuming bootstrap.
+#
+# Use /etc/product/init.d/S01-myscript for scripts, may be a symlink, it
+# will be called with `start` as its only argument.
+#
+# The compatible array is listed in reverse order, least significant to
+# most significant.  We apply all in order so the most significant can
+# override more generic ones.
 ident=$(basename "$0")
 
+PRODUCT_INIT=/etc/product/init.d
 PREFIXD=/usr/share/product
-PRODUCT=$(jq -r '."product-name" | ascii_downcase' /run/system.json)
+COMPATIBLES=$(jq -r '."compatible"[] | ascii_downcase' /run/system.json)
 
 note()
 {
     logger -I $$ -k -p user.notice -t "$ident" "$1"
 }
 
-DIR="$PREFIXD/$PRODUCT"
-if [ -z "$PRODUCT" ] || [ ! -d "$DIR" ]; then
-    note "No vendor/product specific directory found, using built-in defaults."
-    exit 0
+found=false
+for PRODUCT in $COMPATIBLES; do
+    DIR="$PREFIXD/$PRODUCT"
+    if [ -d "$DIR" ]; then
+        note "Using vendor/product-specific defaults for $PRODUCT."
+        for dir in "$DIR"/*; do
+            [ -d "$dir" ] && cp -a "$dir" /
+        done
+        found=true
+    fi
+done
+
+if [ "$found" = false ]; then
+    note "No vendor/product-specific directory found, using built-in defaults."
 fi
 
-note "Using vendor/product specific defaults."
-for dir in "$DIR"/*; do
-    [ -d "$dir" ] && cp -a "$dir" /
-done
+note "Calling runparts $PRODUCT_INIT/S[0-9]+.* start"
+/usr/libexec/finit/runparts -bsp "$PRODUCT_INIT"
+
+# Product specific init done.
+initctl -nbq cond set product
```

Comment:
Not sure if I'm missing something :sweat_smile:, but what I was talking about [here](https://github.com/kernelkit/infix/pull/811/files#r1832704344), was something like this:

```suggestion
# Conditions for bootstrap services, this enables product specific
# init scripts to prevent select services from starting.
initctl -nbq cond set led

note "Calling runparts $PRODUCT_INIT/S[0-9]+.* start"
/usr/libexec/finit/runparts -bsp "$PRODUCT_INIT"

# Product specific init done.
initctl -nbq cond set product
```
...which might make it more clear how `<usr/product,usr/led>` is expected to work, rather than half of it being in another script.

Reply:
Yeah I understand.

To me it was more logical the other way around, the 00-probe script setting all base conditions: led, rauc, mkcert, gdbserver, ... and the 05-product script calling product specific init to potentially clear them, finishing off by setting its own condition.